File: solutions/python/Next Greater Node In Linked List.py
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next
class Solution:
    def nextLargerNodes(self, head: Optional[ListNode]) -> List[int]:
        [stack,arr,temp,i]=[[],[],head,0]
        while temp:
            arr.append(temp.val)
            temp=temp.next;
        for i in range(0,len(arr)):
            while len(stack)>0 and arr[stack[-1]]<arr[i]:
                arr[stack[-1]]=arr[i]
                stack.pop()
            stack.append(i)
        while len(stack)>0:
            arr[stack.pop()]=0
        return arr

File: solutions/Javascript/Next Greater Node In Linked List.js
class ListNode {
  constructor(val, next) {
    this.val = val === undefined ? 0 : val;
    this.next = next === undefined ? null : next;
  }
}

var nextLargerNodes = function (head) {
  let temp = head;
  let stack = [],
    arr = [];
  while (temp) {
    arr.push(temp.val);
    temp = temp.next;
  }
  for (let i = 0; i < arr.length; i++) {
    while (stack.length > 0 && arr[stack[stack.length - 1]] < arr[i]) {
      arr[stack.pop()] = arr[i];
    }
    stack.push(i);
  }
  while (stack.length > 0) {
    arr[stack.pop()] = 0;
  }
  return arr;
};
